File: ios/MenCare/Core/Model/Notifications.swift
import Foundation

struct ReminderRule: Identifiable, Hashable {
    let id: String
    var hoursBefore: Int
}

struct NotificationSettings: Hashable {
    var reminders: [ReminderRule]
    var bookingConfirmation: Bool = true
    var cancellationAlert: Bool = true
    var lateOperatorAlert: Bool = false
    var emptyDayPromos: Bool = false
}

enum CampaignSegment: CaseIterable, Hashable {
    case inattivi60, tutti, topSpesa
}

enum CampaignStatus: Hashable {
    case draft, scheduled, sent
}

struct PushCampaign: Identifiable, Hashable {
    let id: String
    var name: String
    var segment: CampaignSegment
    var title: String
    var body: String
    var scheduledAt: LocalDateTime?
    var repeatWeekly: Bool = false
    var sendCap: Int?
    var reachableCount: Int
    var segmentSize: Int
    var status: CampaignStatus = .draft
}

/// An in-app notification, addressed to one user (client, staff or owner).
struct AppNotification: Identifiable, Hashable {
    let id: String
    let userId: String
    var title: String
    var body: String
    var at: LocalDateTime
    var read: Bool = false
}

File: android/core/model/src/main/kotlin/com/devora/mencare/core/model/Notifications.kt
package com.devora.mencare.core.model

import java.time.LocalDateTime

data class ReminderRule(
    val id: String,
    val hoursBefore: Int,
)

data class NotificationSettings(
    val reminders: List<ReminderRule>,
    val bookingConfirmation: Boolean = true,
    val cancellationAlert: Boolean = true,
    val lateOperatorAlert: Boolean = false,
    val emptyDayPromos: Boolean = false,
)

enum class CampaignSegment { INATTIVI_60, TUTTI, TOP_SPESA }

enum class CampaignStatus { DRAFT, SCHEDULED, SENT }

data class PushCampaign(
    val id: String,
    val name: String,
    val segment: CampaignSegment,
    val title: String,
    val body: String,
    val scheduledAt: LocalDateTime? = null,
    val repeatWeekly: Boolean = false,
    val sendCap: Int? = null,
    val reachableCount: Int,
    val segmentSize: Int,
    val status: CampaignStatus = CampaignStatus.DRAFT,
)

/** An in-app notification, addressed to one user (client, staff or owner). */
data class AppNotification(
    val id: String,
    val userId: String,
    val title: String,
    val body: String,
    val at: LocalDateTime,
    val read: Boolean = false,
)
